index.js

```diff
@@ -285,28 +365,35 @@ studio.protocol = (function(ProtoBuf) {
     this.handle = function(message){
       return new Promise(function(resolve, reject) {
         try {
-          var hello = obj.Hello.decode(message);
+          var hello = obj.Hello.decode(new Uint8Array(message));
         } catch (err) {
           console.log("Hello Error: "+err+"\n");
           onError();
           resolve(new ErrorHandler());
+          return;
         }
 
         function applicationAcceptanceRequested(request){
           return new Promise(function(resolve, reject) {
-            if (request.systemUseNotification())
-              window.confirm(metadata.systemUseNotification) ? resolve() : reject();
-            else
+            if (request.systemUseNotification()) {
+              if (typeof window !== 'undefined' && window.confirm) {
+                window.confirm(metadata.systemUseNotification) ? resolve() : reject();
+              } else {
+                resolve();
+              }
+            } else {
               resolve();
+            }
           });
         }
 
         var metadata = {}
-        metadata.systemName = hello.system_name;
-        metadata.applicationName = hello.application_name;
-        metadata.cdpVersion = hello.cdp_version_major + '.' + hello.cdp_version_minor + '.' + hello.cdp_version_patch;
-        metadata.systemUseNotification = hello.system_use_notification;
+        metadata.systemName = hello.systemName;
+        metadata.applicationName = hello.applicationName;
+        metadata.cdpVersion = hello.cdpVersionMajor + '.' + hello.cdpVersionMinor + '.' + hello.cdpVersionPatch;
+        metadata.systemUseNotification = hello.systemUseNotification;
         metadata.challenge = hello.challenge;
+        metadata.compatVersion = hello.compatVersion || hello.compat_version || 2;
```

Comment:
Should decide whether it is compatVersion or compat_version. And I don't think it is correct to assign a default value here